hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -257,6 +257,14 @@ public static IOException translateException(@Nullable String operation,
         ioe = new AWSBadRequestException(message, s3Exception);
         break;
 
+      // version/etag id cannot be met in copy.
+      case 412:
+        ioe = new RemoteFileChangedException(path,
+            operation,
+            RemoteFileChangedException.PRECONDITIONS_NOT_MET,
+            ase);
+        break;
+
       // out of range. This may happen if an object is overwritten with
       // a shorter one while it is being read.
       case 416:
```

Comment:
RANGE_NOT_SATISFIABLE

Reply:
(argh, hitting cmd-enter turns this into a review and not a comment like it used to)